### PR TITLE
Add thin arbiter support for virtual block

### PR DESCRIPTION
**Pull request by kotreshhr**
Adds thin arbiter block hosting volume support for virtual block.
A storageclass is provided thin arbiter virtual block.
Any PVC requested with storageclass of thin arbiter virtual block
will be served from block hosting volume of type thin arbiter.

Signed-off-by: Kotresh HR <khiremat@redhat.com>

### CODE REVIEW

**Comment by kotreshhr** [comment]
@Madhu-1 @aravindavk @poornimag @amarts PTAL

File: examples/kubernetes/thin-arbiter-pod.yaml [hunk 1]
```diff
@@ -1,19 +0,0 @@
----
```
Comment (Madhu-1) [inline examples/kubernetes/thin-arbiter-pod.yaml]: why these examples are removed?

File: pkg/gluster-virtblock/controllerserver.go [hunk 2]
```diff
@@ -43,11 +45,90 @@ type CsiDrvParam struct {
...
+func setReplicaCnt(volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	replicaCnt, err := utils.ParseVolumeParamInt("replicas", value)
+	if err != nil {
+		return err
+	}
+	volumeReq.HostVolumeInfo.HostVolReplicaCnt = replicaCnt
+	return nil
+}
+func setArbiterReplicaCnt(req *csi.CreateVolumeRequest, volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	if value == "thin" {
```
Comment (Madhu-1) [inline pkg/gluster-virtblock/controllerserver.go]:
```
if value == "thin" {
		if err := utils.ValidateThinArbiter(req); err != nil {
			return err
		}
		volumeReq.HostVolumeInfo.HostVolReplicaCnt = 2
               return nil
} 
return fmt.Errorf("invalid arbiterType specified: %s", value)
```

File: pkg/gluster-virtblock/controllerserver.go [hunk 3]
```diff
@@ -43,11 +45,90 @@ type CsiDrvParam struct {
...
+	} else {
+		return fmt.Errorf("invalid arbiterType specified: %s", value)
+	}
+	return nil
+}
+
+func setArbiterPath(req *csi.CreateVolumeRequest, volumeReq *bapi.BlockVolumeCreateRequest, value string) error {
+	if _, ok := req.Parameters["arbiterType"]; !ok {
+		glog.Error("only arbiterPath provided, missing arbiterType")
+		return fmt.Errorf("only arbiterPath provided, missing arbiterType")
```
Comment (Madhu-1) [inline pkg/gluster-virtblock/controllerserver.go]: use Errors.New()

File: pkg/gluster-virtblock/controllerserver.go [hunk 4]
```diff
@@ -43,11 +45,90 @@ type CsiDrvParam struct {
...
+		switch k {
+		case "replicas":
+			err = setReplicaCnt(volumeReq, v)
+		case "arbiterType":
+			err = setArbiterReplicaCnt(req, volumeReq, v)
+		case "arbiterPath":
+			err = setArbiterPath(req, volumeReq, v)
+		case "shardSize":
+			err = setShardSize(volumeReq, v)
+		case "hostVolumeSize":
```
Comment (Madhu-1) [inline pkg/gluster-virtblock/controllerserver.go]: need to add this value to storage-class example and please  update readme what this value  meant to be

File: pkg/utils/common-utils.go [hunk 5]
```diff
@@ -1,12 +1,16 @@
 package utils
 
 import (
+	"errors"
+	"fmt"
```
Comment (Madhu-1) [inline pkg/utils/common-utils.go]: import orders are not  proper

File: pkg/utils/common-utils.go [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
```
Comment (Madhu-1) [inline pkg/utils/common-utils.go]: need new line here

**Pull request merged**